src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,50 +1,109 @@
 import { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (newMovie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setimgUrl] = useState('');
+  const [imdbUrl, setimdbUrl] = useState('');
+  const [imdbId, setimdbId] = useState('');
+
+  const clear = () => {
+    setTitle('');
+    setDescription('');
+    setimgUrl('');
+    setimdbUrl('');
+  };
+
+  const isEnabled = title && imgUrl && imdbUrl && imdbId;
+
+  const handleSubmit = () => {
+    const newMovie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    onAdd(newMovie);
+    setCount(current => current + 1);
+    clear();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+        handleSubmit();
+      }}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(event) => {
+          setTitle(event); // why this way
+        }}
```

Comment:
your question is spot-on here.
that might be a default behaviour for `onChange`
so it does work as-is, and you can use a shorthand syntax
```suggestion
        onChange={setTitle}
```

yet of course it's better to remain explicit
```
(event) => setState(event.target.value)
```